search/application/searchService.go

```diff
@@ -55,16 +72,16 @@ func (s *SearchService) FindBy(ctx context.Context, documentType string, searchR
 	}
 
 	if searchRequest.PaginationConfig == nil {
-		searchRequest.PaginationConfig = s.PaginationInfoFactory.DefaultConfig
+		searchRequest.PaginationConfig = s.paginationInfoFactory.DefaultConfig
 	}
 
 	// pageSize can either be set in the request, or we use the configured default or if nothing set we rely on the ProductSearchService later
 	pageSize := searchRequest.PageSize
 	if pageSize == 0 {
-		pageSize = int(s.DefaultPageSize)
+		pageSize = int(s.defaultPageSize)
 	}
 
-	result, err := s.SearchService.SearchFor(ctx, documentType, BuildFilters(searchRequest, pageSize)...)
+	result, err := s.searchService.SearchFor(ctx, documentType, BuildFilters(searchRequest, pageSize)...)
```

Comment:
since searchService is optional now, we should insert guard clauses to prevent nil pointer dereferences

Reply:
Thanks make sense - added guard